### PR TITLE
feat(metering): add metering for logs

### PR DESCRIPTION
**Pull request by grandwizard28**
#### Features

- Add a pkg `metering` with interface `metering.Meter` for Size and Count usage calculation
- Add a pkg `pdatagen` for generating a wide variety of telemetry data for testing

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]

**Comment by shivanshuraj1333** [comment]
```
=== RUN   TestLogsSize
    logs_test.go:24: 
        	Error Trace:	/Users/s/oss/sig/signoz-otel-collector/pkg/metering/v1/logs_test.go:24
        	Error:      	Not equal: 
        	            	expected: 3890
        	            	actual  : 3910
        	Test:       	TestLogsSize
--- FAIL: TestLogsSize (0.00s)

Expected :3890
Actual   :3910
<Click to see difference>


FAIL

Process finished with the exit code 1
```

**Review by srikanthccv — APPROVED** [review]

**Comment by shivanshuraj1333** [comment]
We should have this package in the gateway-collector as I don't think it is going to be used in the signoz-collector

See, https://signoz-team.slack.com/archives/C050MERV57W/p1735396691630329?thread_ts=1735138328.325539&cid=C050MERV57W for more details.

File: pkg/metering/v1/logs.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package v1
+
+import (
+	"github.com/SigNoz/signoz-otel-collector/pkg/metering"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type logs struct {
+	Logger *zap.Logger
```
Comment (shivanshuraj1333) [inline pkg/metering/v1/logs.go:10]: This is redundant and not being use anywhere in the code.

File: pkg/metering/v1/logs.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	Sizer  metering.Sizer
+}
+
+func NewLogs(logger *zap.Logger) metering.Logs {
+	return &logs{
+		Logger: logger,
+		Sizer:  metering.NewJSONSizer(logger),
```
Comment (shivanshuraj1333) [inline pkg/metering/v1/logs.go:17]: this is hardcoded for `json`, in the gateway, through the encoder one can also configure `proto`, https://github.com/SigNoz/opentelemetry-gateway/blob/main/internal/processor/signozlimiterprocessor/processor.go#L91

**Pull request merged**